ci/post/main.py

```diff
@@ -62,22 +62,41 @@ def _match_version_number(text, regex):
 	return int(match.group(1))
 	
 
-def get_source_version(date_version):
+def get_source_version(date_version: datetime, tag_name: str) -> semantic_version.Version:
 	"""! Retrieves the build's version from `version.cmake`, forms it as a string, and appends the date
 
 	@param[in] `date_version` Date this build is published for release
+	@param[in] 'tag_name'     Tag of this release, used to determine if release, rc, or nightly
 
-	@return String of the form `MAJOR_VERSION.MINOR_VERSION.BUILD_VERSION-date`
+	@return String of the form `MAJOR_VERSION.MINOR_VERSION.BUILD_VERSION-date`, ' MAJOR_VERS
 	"""
-	
+	major = minor = build = version = 0
 	with open(os.path.join("..", "..", "cmake", "version.cmake"), "r") as f:
 		filetext = f.read()
-
 		major = _match_version_number(filetext, MAJOR_VERSION_PATTERN)
 		minor = _match_version_number(filetext, MINOR_VERSION_PATTERN)
 		build = _match_version_number(filetext, BUILD_VERSION_PATTERN)
 
-		return semantic_version.Version("{}.{}.{}-{}".format(major, minor, build, date_version))
+	if "rc" in tag_name.lower():
+		# Release canidates retain an internal version unstable, but external version is the tagname
```

Comment:
canidates -> candidates

also what does "an internal version unstable" mean?

Reply:
This is supposed to document that release candidates still maintain their unstable/nightly revision scheme, but the .exe name uses the <major>.<minor>.<build>-<RCn> form